opennms-services/src/test/java/org/opennms/netmgt/collectd/SnmpCollectorMinMaxValIT.java

```diff
@@ -238,9 +240,10 @@ public void testPersist() throws Exception {
 
         // by now the values should be the new values
         assertEquals(Double.valueOf(456.0), m_rrdStrategy.fetchLastValueInRange(rrdFile.getAbsolutePath(), "tcpCurrEstab", stepSizeInMillis, rangeSizeInMillis));
+        // TODO: Patrick: is this correct?
```

Comment:
@j-white I am not sure that this test is correct? We set the value 7654321 in line 235. But in line 242 we test for  the old value _1234567.0_

I think the reason why this test passed with the old implementation is that the value was never set:
1. Block value = 1234567
1. Block value = 7654321 I don't think this gets set, it seems that we can only increment by one
1. Block value = 1234567 
It could also be that I am not clear on what we are testing here but it seems strange to me. I did a bit more investigation and wrote the following tests:
https://github.com/OpenNMS/opennms/pull/2290/files#diff-a64257aa8c4f16de0a23885e4a4fd2c3R274
https://github.com/OpenNMS/opennms/pull/2290/files#diff-f2c4541f8f65108cc68347ab00d15adeR273

It looks like that we can set any value in the new Implementation but only increment by one in the old implementation. Is the old behavior the desired one? Could you help me to understand what is going on? Thanks!

Reply:
We figure this out. The test enforces a max value which explains the observed behavior.